lib/fqn/fqn.go

```diff
@@ -0,0 +1,250 @@
+func ParseAttributeFqn(fqn string) (*FullyQualifiedAttribute, error) {
+	parsed := &FullyQualifiedAttribute{
+		Fqn: fqn,
+	}
+
+	// First try to match against the attribute value pattern
+	valueMatches := attributeValueFQNRegex.FindStringSubmatch(fqn)
+	if len(valueMatches) > 0 {
+		namespaceIdx := attributeValueFQNRegex.SubexpIndex("namespace")
+		nameIdx := attributeValueFQNRegex.SubexpIndex("name")
+		valueIdx := attributeValueFQNRegex.SubexpIndex("value")
+
+		if len(valueMatches) <= namespaceIdx || len(valueMatches) <= nameIdx || len(valueMatches) <= valueIdx {
+			return nil, fmt.Errorf("%w: valid attribute value FQN format https://<namespace>/attr/<name>/value/<value> must be provided", ErrInvalidFQNFormat)
+		}
+
+		ns := valueMatches[namespaceIdx]
+		name := valueMatches[nameIdx]
+		value := valueMatches[valueIdx]
+
+		isValid := validNamespaceRegex.MatchString(ns) && validObjectNameRegex.MatchString(name) && validObjectNameRegex.MatchString(value)
+		if !isValid {
+			return nil, fmt.Errorf("%w: found namespace %s with attribute name %s and value %s", ErrInvalidFQNFormat, ns, name, value)
+		}
+
+		parsed.Namespace = ns
+		parsed.Name = name
+		parsed.Value = value
+
+		return parsed, nil
+	}
+
+	// If not a value FQN, try to match against the attribute definition pattern
+	defMatches := attributeDefinitionFQNRegex.FindStringSubmatch(fqn)
+	if len(defMatches) > 0 {
+		namespaceIdx := attributeDefinitionFQNRegex.SubexpIndex("namespace")
+		nameIdx := attributeDefinitionFQNRegex.SubexpIndex("name")
+
+		if len(defMatches) <= namespaceIdx || len(defMatches) <= nameIdx {
+			return nil, errors.New("error: valid attribute definition FQN format https://<namespace>/attr/<name> must be provided")
+		}
+
+		ns := defMatches[namespaceIdx]
+		name := defMatches[nameIdx]
+
+		isValid := validNamespaceRegex.MatchString(ns) && validObjectNameRegex.MatchString(name)
+		if !isValid {
+			return nil, fmt.Errorf("%w: found namespace %s with attribute name %s", ErrInvalidFQNFormat, ns, name)
+		}
+		parsed.Namespace = ns
+		parsed.Name = name
+
+		return parsed, nil
+	}
+
+	// If not a definition FQN, try to match against just the namespace
+	nsMatches := namespaceOnlyRegex.FindStringSubmatch(fqn)
+	if len(nsMatches) > 0 {
+		namespaceIdx := namespaceOnlyRegex.SubexpIndex("namespace")
+
+		if len(nsMatches) <= namespaceIdx {
+			return nil, errors.New("error: valid namespace FQN format https://<namespace> must be provided")
+		}
+
+		ns := nsMatches[namespaceIdx]
+		isValid := validNamespaceRegex.MatchString(ns)
+		if !isValid {
+			return nil, fmt.Errorf("%w: found namespace %s", ErrInvalidFQNFormat, ns)
+		}
+
+		parsed.Namespace = ns
+		return parsed, nil
+	}
+
+	return nil, errors.New("error: invalid attribute FQN format, must be https://<namespace>, https://<namespace>/attr/<name>, or https://<namespace>/attr/<name>/value/<value>")
+}
+
+func AttributeFqnBuilder(namespace, name, value string) (string, error) {
```

Comment:
```suggestion
func AttributeBuilder(namespace, name, value string) (string, error) {
```

Also shouldn't this return AttributeValue or something other than `string`?